ui/appui-react/src/appui-react/popup/ComponentPopup.tsx

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import type { Orientation, SizeProps } from "@itwin/core-react";
+import { DivWithOutsideClick, Point, Size } from "@itwin/core-react";
+import * as React from "react";
+import { CursorPopup } from "../cursor/cursorpopup/CursorPopup";
+import type { PopupPropsBase } from "./PopupManager";
+import { PopupManager } from "./PopupManager";
+import { PositionPopup } from "./PositionPopup";
+import type { OnCancelFunc, RelativePosition } from "@itwin/appui-abstract";
+import { useEffect, useState } from "react";
+import { WrapperContext } from "../configurableui/ConfigurableUiContent";
+
+/**
+ *
+ */
+export interface ComponentPopupProps extends Omit<PopupPropsBase, "el"> {
+  Component: React.ReactElement;
+  relativePosition: RelativePosition;
+  orientation: Orientation;
+  onCancel: OnCancelFunc;
+  anchorRef?: React.RefObject<HTMLElement>;
+}
+
+/**
+ * @alpha
+ */
+export const ComponentPopup: React.FC<ComponentPopupProps> = ({
```

Comment:
Do we want/need this exported? What is the use-case for this component (missing docs)? Can we think of a better name?

Reply:
We do need it exported so it can be used by the `PopupManager`, unless we want to inline it over there. Name-wise, it's a bit tough. What the component does is wrap and position a component as a "popup." But there's already a component `PositionPopup` which could get confusing. 

As discussed, all of these and the PopupManager/Renderer could be revisited - there's a fair amount of shared functionality, but not sure this is the best PR to do it in.  

Doc added. Set access level to @internal.

Reply:
Thanks, I meant access level (release tag).